gaeguli/pipeline.c

```diff
@@ -164,10 +168,26 @@ gaeguli_pipeline_dispose (GObject * object)
         "GaeguliPipeline reference!");
   }
 
+  if (!self->loop || !g_main_loop_is_running (self->loop)) {
+    g_main_loop_quit (self->loop);
+  }
+
+  g_clear_pointer (&self->loop, g_main_loop_unref);
+  g_clear_pointer (&self->context, g_main_context_unref);
+
+  if (!self->thread) {
+    if (self->thread != g_thread_self ())
+      g_thread_join (self->thread);
+    else
+      g_thread_unref (self->thread);
+    self->thread = NULL;
```

Comment:
`thread` is NULL within this scope.

Reply:
I tended to set `NULL` to `thread` after `unref()`. Do you mean using `{` and `}` for if-statement?

Reply:
I mean this whole block is inside `if (!self->thread) { do-something-with-my-NULL-thread }`

Reply:
ah... you pointed inverted mixup. Thank you!